src/minichain/transaction.py

```
@@ -0,0 +1,144 @@
+        if not isinstance(self.nonce, int) or self.nonce < 0:
+            return False
+        if not isinstance(self.fee, int) or self.fee < 0:
+            return False
+        if not isinstance(self.timestamp, int) or self.timestamp < 0:
+            return False
+        return True
+
+    def sign(self, signing_key: object) -> None:
+        """Sign this transaction in-place and populate auth fields."""
+        if not self._validate_common_fields():
+            raise ValueError("Invalid transaction fields")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)